### PR TITLE
UI Migration from map.js to map.html

### LINKED ISSUE #118: Migrate Map topbar interactions out of Map.js and into map.HTML ("map UI Migration")
This way we can save on some complexity in map.js, and also not have to create/code our own functions for mouse over/input/clicks etc.

### PR DESCRIPTION
**Pull request by Muggyfox1**
Moved majority of UI code/functionality to be in HTML rather than JavaScript.
resolves #118

### CODE REVIEW

File: data/quest.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
 	]}
     ,{"name":"Journal-Filled","elements":[
 	    {"name":"Cast out of the Thieves Guild","formId":"0x0001E868","notes":"Do this quest once before you finish the Thieves Guild. You can never finish this quest, but you need to get it in your journal for true 100%."}
-       ,{"name":"Expelled from the Fighters Guild","formId":"0x00025E27","notes":"Do this quest once before you finish the Thieves Guild. You can never finish this quest, but you need to get it in your journal for true 100%."}
+       ,{"name":"Expelled from the Fighters Guild","formId":"0x00025E27","notes":"Do this quest once before you finish the Fighters Guild. You can never finish this quest, but you need to get it in your journal for true 100%."}
```
Comment (Muggyfox1) [inline data/quest.json:25]: Bosch pointed out this typo, just fixed it here out of convenience.

File: map.html [hunk 2]
```diff
@@ -73,9 +73,15 @@
 </div>
 <br/>
 
-<h1 style="text-align: center;">Map Area</h1>
-<p>This is where the explanation goes.</p>
-<div style="height:80vh;" id="wrapper_Map"></div>
+
+<div style="margin-top: 1em;">
+	<button id="button_Location" title="Shows all Locations that need to be discovered.">Locations</button>
```
Comment (Muggyfox1) [inline map.html:78]: These are plain buttons, do we want to spice it up to be styled like the top bar?

File: map.html [hunk 3]
```diff
@@ -73,9 +73,15 @@
...
-<div style="height:80vh;" id="wrapper_Map"></div>
+
+<div style="margin-top: 1em;">
+	<button id="button_Location" title="Shows all Locations that need to be discovered.">Locations</button> 
+	<button id="button_Nirnroot" title="Shows where all Nirnroot are in the overworld.">Nirnroots</button>
+	<button id="button_ToggleTSP" title="The Traveling Salesman Path(TSP) is the fastest way to visit each location/nirnroot">Toggle Traveling Salesman Path</button>
+	<button onclick="showHelp()">Help!</button>
+</div>
+
+<div style="height:85vh; border: solid 1px" id="wrapper_Map"></div>
```
Comment (Muggyfox1) [inline map.html:84]: increased map height since we don't have as much space taken up by a header.

File: map.html [hunk 4]
```diff
@@ -91,5 +97,20 @@ <h1 style="text-align: center;">Map Area</h1>
...
+	let welcome = "Welcome to the map!\n\n";
+	let clickIcon = "Click on an icon when it's been discovered/picked.\n\n";
+	let tspExplanation = "The Traveling Salesman Path is a calculated fastest route between each icon, it doesn't have to be done in order and it loops so you can start anywhere, and as long as you only go one direction, you will get everything on the path.\n\n";
+	
+	//this isn't implemented yet, but should be noted so people aren't surprised when it happens during their run.
+	let randomGateNote = "You need to only discover 40 random oblivion gates. Once you discover 40, the map will stop showing the remaining random gate locations."
+	
+	let helpText = welcome + clickIcon + tspExplanation + randomGateNote;
+	function showHelp(){
+		alert(helpText);
```
Comment (Muggyfox1) [inline map.html:112]: Probably not the best practice for getting a large string explanation put together, just wanted to throw it in this way so I could edit it easier.

other ideas would be just type it in straight on alert() or use an array of strings.

File: map.js [hunk 5]
```diff
@@ -1,13 +1,11 @@
 //TODO: Figure out Random Gate tracking?
-    //seperate counter for found random gates?
+    //seperate counter for found random gates, count to 40 and stop drawing remaining gates.
 
-//some decorations from the guide that might be useful.
-    //TODO: Add pink circle for fixed gates 
+//some decorations from the guide that might be useful. //or We could just insert these things into the name of the gates.
```
Comment (Muggyfox1) [inline map.js:4]: Thinking about this more, we should probably put this info instead on the json oblivion gate entries.

File: map.html (hunk 3 above)
Comment (MichaelEbert) [reply map.html:84]: hmm. ideally we would just auto format it to the rest of the window size. I'll see about doing a ui rewrite soon.

File: map.html (hunk 4 above)
Comment (MichaelEbert) [reply map.html:112]: I love it.

File: map.js (hunk 5 above)
Comment (MichaelEbert) [reply map.js:4]: yep. it's a Hard Problem and, like fame, will probably take me way too long to implement.

**Review by MichaelEbert — APPROVED** [review]
LGTM

**Pull request merged**